packages/devtools_app/lib/src/screens/performance/panes/queued_microtasks/queued_microtasks_view.dart

```diff
@@ -0,0 +1,278 @@
+import 'package:intl/intl.dart' show DateFormat;
+import 'package:vm_service/vm_service.dart';
+
+import '../../../../shared/analytics/constants.dart' as gac;
+import '../../../../shared/primitives/utils.dart' show SortDirection;
+import '../../../../shared/table/table.dart' show FlatTable;
+import '../../../../shared/table/table_data.dart';
+import '../../../../shared/ui/common_widgets.dart' show RefreshButton;
+import 'queued_microtasks_controller.dart';
+
+class RefreshQueuedMicrotasksButton extends StatelessWidget {
+  const RefreshQueuedMicrotasksButton({
+    super.key,
+    required QueuedMicrotasksController controller,
+  }) : _controller = controller;
+
+  final QueuedMicrotasksController _controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<QueuedMicrotasksControllerStatus>(
+      valueListenable: _controller.status,
+      builder: (_, status, _) {
+        return RefreshButton(
+          iconOnly: true,
+          outlined: false,
+          onPressed: status == QueuedMicrotasksControllerStatus.refreshing
+              ? null
+              : _controller.refresh,
+          tooltip:
+              "Take a new snapshot of the selected isolate's microtask queue.",
+          gaScreen: gac.performance,
+          gaSelection: gac.PerformanceEvents.refreshQueuedMicrotasks.name,
+        );
+      },
+    );
+  }
+}
+
+class QueuedMicrotasksTabControls extends StatelessWidget {
+  const QueuedMicrotasksTabControls({
+    super.key,
+    required QueuedMicrotasksController controller,
+  }) : _controller = controller;
+
+  final QueuedMicrotasksController _controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [RefreshQueuedMicrotasksButton(controller: _controller)],
+    );
+  }
+}
+
+class RefreshQueuedMicrotasksInstructions extends StatelessWidget {
+  const RefreshQueuedMicrotasksInstructions({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: RichText(
+        text: TextSpan(
+          style: Theme.of(context).regularTextStyle,
+          children: [
+            const TextSpan(text: 'Click the refresh button '),
+            WidgetSpan(child: Icon(Icons.refresh, size: defaultIconSize)),
+            const TextSpan(
+              text:
+                  " to take a new snapshot of the selected isolate's "
+                  'microtask queue.',
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+// In the response returned by the VM Service, microtasks are sorted in
+// ascending order of when they will be dequeued, i.e. the microtask that will
+// run earliest is at index 0 of the returned list. We use those indices of the
+// returned list to sort the entries of the microtask selector, so that they
+// they also appear in ascending order of when they will be dequeued.
+typedef IndexedMicrotask = (int, Microtask);
```

Comment:
can we used named elements in this record so that we don't have to use `$1` and `$2`?